contracts/staking/slashing/SlashingManager.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity ^0.5.2;
+
+import {Ownable} from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import {SafeMath} from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import {RLPReader} from "solidity-rlp/contracts/RLPReader.sol";
+import {IERC20} from "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+import {StakeManager} from "../stakeManager/StakeManager.sol";
+import {Registry} from "../../common/Registry.sol";
+import {StakingInfo} from "../StakingInfo.sol";
+import "./ISlashingManager.sol";
+
+
+contract SlashingManager is ISlashingManager, Ownable {
+    using SafeMath for uint256;
+    using RLPReader for bytes;
+    using RLPReader for RLPReader.RLPItem;
+
+    modifier onlyStakeManager() {
+        require(registry.getStakeManagerAddress() == msg.sender);
+        _;
+    }
+
+    constructor(address _registry, address _logger) public {
+        registry = Registry(_registry);
+        logger = StakingInfo(_logger);
+    }
+
+    function updateSlashedAmounts(
+        address proposer,
+        bytes memory vote,
+        bytes memory sigs,
+        bytes memory slashingInfoList,
+        bytes memory txData
+    ) public {
+        RLPReader.RLPItem[] memory dataList = vote.toRlpItem().toList();
+        require(
+            keccak256(dataList[0].toBytes()) == heimdallId,
+            "Chain ID is invalid"
+        );
+        require(dataList[1].toUint() == VOTE_TYPE, "Vote type is invalid");
+        require(slashingNonce < dataList[2].toUint(), "Invalid slashing nonce");
+        slashingNonce = dataList[2].toUint();
+        StakeManager stakeManager = StakeManager(
+            registry.getStakeManagerAddress()
+        );
+
+        require(
+            keccak256(dataList[4].toBytes()) ==
+                keccak256(abi.encodePacked(sha256(txData))),
+            "Extra data is invalid"
+        );
+
+        uint256 stakePower;
+        uint256 activeTwoByThree;
+        (stakePower, activeTwoByThree) = logger.verifyConsensus(
+            keccak256(vote),
+            sigs
+        );
+        dataList = txData.toRlpItem().toList()[0].toList();
+        require(stakePower >= activeTwoByThree, "2/3+1 Power required");
+        require(dataList[0].toAddress() == proposer, "Invalid proposer");
+        require(
+            keccak256(dataList[1].toBytes()) ==
+                keccak256(abi.encodePacked(sha256(slashingInfoList))),
+            "Invalid slashInfoHash"
+        );
+        //slashingInfoList[]=[[valiD,am,isJailed]]
+        uint256 slashedAmount = stakeManager.slash(slashingInfoList);
+        uint256 bounty = (slashedAmount.mul(reportRate)).div(100);
+        slashedAmount = slashedAmount.sub(bounty);
+        require(
+            stakeManager.transferFunds(
```

Comment:
To avoid duplication, might be a good idea to place `stakeManager.transferFunds` in an internal function and call that everywhere.

Reply:
sorry I didn't get😐

Reply:
```
function _stakeFundsTransfer() {
require(
            stakeManager.transferFunds(0,
                slashedAmount,
                address(this)
            ),
            "Transfer failed"
        );
```
And then use `_stakeFundsTransfer`, everywhere.

Reply:
sorry I still don't get the point because I would still need to call `_stakeFundsTransfer` 3 times with different params. which will do the same thing as the current one eventually.